utility/utils.py

```diff
@@ -0,0 +1,134 @@
+"""
+General utility functions 
+
+"""
+import logging
+import os
+import uuid
+import hashlib
+
+from framework import config
+from common_ci_utils.command_runner import exec_cmd
+from framework.ssh_connection_manager import SSHConnectionManager
+
+log = logging.getLogger(__name__)
+
+
+def get_noobaa_sa_host_home_path():
+    """
+    Get the full path of the home directory on the remote machine
+
+    Returns:
+        str: The full path of the home directory on the remote machine
+
+    """
+    cmd = "echo $HOME"
+    _, stdout, _ = SSHConnectionManager().connection.exec_cmd(cmd)
+    return stdout
+
+
+def generate_random_hex(length=20):
+    """
+    Generates a random hexadecimal string with the given length
+
+    Args:
+        length (int): The length of the hexadecimal string.
+
+    Returns:
+        str: A random hexamiadecimal string.
+
+    """
+    cmd = f"openssl rand -hex {length}"
+    completed_process = exec_cmd(cmd)
+    stdout = completed_process.stdout
+    return stdout.decode("utf-8").strip()
+
+
+def generate_unique_resource_name(prefix="resource"):
+    """
+    Generates a unique resource name with the given prefix
+
+    Args:
+        prefix (str): The prefix of the resource name.
+
+    Returns:
+        str: The unique resource name.
+
+    """
+    unique_id = str(uuid.uuid4()).split("-")[0]
+    return f"{prefix}-{unique_id}"
+
+
+def get_current_test_name():
+    """
+    Get the name of the current test
+
+    Returns:
+        str: The name of the current PyTest test
+
+    """
+    return os.environ.get("PYTEST_CURRENT_TEST").split(":")[-1].split(" ")[0]
+
+
+def get_config_root_full_path():
+    """
+    Get the full path of the configuration root directory on the remote machine
+
+    Returns:
+        str: The full path of the configuration root directory on the remote machine
+
+    """
+    config_root = config.ENV_DATA["config_root"]
+
+    if config_root.startswith("~/") == False:
+        return config_root
+
+    config_root = config_root.split("~/")[1]
+    return f"{get_noobaa_sa_host_home_path()}/{config_root}"
+
+
+def generate_random_files(dir, amount, size):
+    """
+    Generate random files in a given directory
+
+    Args:
+        dir (str): The directory in which to generate the files
+        amount (int): The number of files to generate
+        size (str): The size of each file
+
+    Returns:
+        list: A list of the files generated
+
+    """
+    log.info(f"Generating {amount} files of size {size} in {dir}")
```

Comment:
I'm not sure if I follow your exact request here. Do you mean to add a flag that would signal to generate random sized objects that are smaller than `size`, or to add an argument that accepts a list of requested sizes? A code example for the requested usage would be the best way to clarify. 

Either way, I think we should add such an enhancement in a follow up PR.

Reply:
Yes... I mean to add a flag that would signal to generate random sized objects that are smaller than size

Reply:
Done in https://github.com/red-hat-storage/noobaa-sa-ci/pull/11/commits/6dcd5fe0e6c52a51438afbc944abd3be929cfae2: instead of adding a signal argument, I split the `size` argument into `min_size` and `max_size`:
```
    def put_random_objects(
        self,
        bucket_name,
        amount=1,
        min_size="1M",
        max_size="1M",
        prefix="",
        files_dir="",
    ):
```

This way if you set them to be the same size then you will upload X objects of the same size, but if you use min_size=5K and max_size=10K you will generate random files of random sizes in this inclusive range.

Reply:
Fixed edge case handling and added a bit more documentation in https://github.com/red-hat-storage/noobaa-sa-ci/pull/11/commits/083d1929cc8b0233ead30fe274a50fce2af1231f